### PR TITLE
Logging: set file to read-only & add logging mode option

### PR DESCRIPTION
**Pull request by bkueng**
This PR includes 2 changes:
- change the log file permissions to read-only after the file is closed. This can be used by other clients to detect when logging is finished
- add an option `LogMode` that can be set to `arm_until_disarm`. In that case logging is started when the vehicle is armed and stopped when disarmed. The default is `always`, which is the same behavior as before.

### CODE REVIEW

File: examples/config.sample [hunk 1]
```diff
@@ -31,6 +31,11 @@
 #       Path to directory where to store flightstack log.
 #       No default value. If absent, no flightstack log will be stored.
 #
+#   LogMode
+#       One of <always>, <arm_until_disarm>
```
Comment (lucasdemarchi) [inline examples/config.sample]: always / while-armed would be better names imo

File: examples/config.sample [hunk 2]
```diff
@@ -31,6 +31,11 @@
...
+#       Default: always. If absent, log from start until mavlink-router is
+#       stopped.
```
Comment (lucasdemarchi) [inline examples/config.sample]: also document the other one, mentioning the log file will be open when the vehicle is armed and closed when disarmed, creating a new log file.

File: examples/config.sample [hunk 3]
```diff
@@ -98,8 +103,7 @@
 #Mavlink-router serves on this TCP port
 TcpServerPort=5790
 ReportStats=false
-#Which mavlink dialect is being used. Can be `common`[default] or `ardupilotmega`
-MavlinkDialect=ardupilotmega
+MavlinkDialect=auto
```
Comment (lucasdemarchi) [inline examples/config.sample]: unrelated. Either a separate commit or at least mention in the commit message that while changing this you fixed the default value here.

File: src/mavlink-router/binlog.cpp [hunk 4]
```diff
@@ -75,22 +75,31 @@ int BinLog::write_msg(const struct buffer *buffer)
...
         struct mavlink_router_mavlink1_header *msg
             = (struct mavlink_router_mavlink1_header *)buffer->data;
         msg_id = msg->msgid;
+        payload = buffer->data + sizeof(struct mavlink_router_mavlink1_header);
         payload_len = msg->payload_len;
+        source_system_id = msg->sysid;
     }
 
+    /* Check if we should start or stop logging */
+    _handle_auto_start_stop(msg_id, source_system_id, payload);
```
Comment (lucasdemarchi) [inline src/mavlink-router/binlog.cpp]: I don't like how this is a special case and the routing to the logging endpoint is plugged in the middle of this function. What's the problem with letting it inside the logendpoint itself implementing the normal pattern like other endpoints with accept_msg() + write()?

Comment (bkueng) [reply src/mavlink-router/binlog.cpp]: I agree it's not that nice. But from what I see `accept_msg` cannot filter by `source_system_id` which is what we want here.

**Comment by bkueng** [comment]
Thanks for the review, I addressed all comments except for `_handle_auto_start_stop`.

**Comment by lucasdemarchi** [comment]
while_armed should be while-armed, but this can be polished on top.

**Comment by lucasdemarchi** [comment]
Applied, thanks.

**Pull request merged**